src/solvers/fd_tree/fd_2d.jl

```diff
@@ -0,0 +1,199 @@
+
+# General interface methods for SummationByPartsOperators.jl and Trixi.jl
+nnodes(D::AbstractDerivativeOperator) = size(D, 1)
+eachnode(D::AbstractDerivativeOperator) = Base.OneTo(nnodes(D))
+get_nodes(D::AbstractDerivativeOperator) = grid(D)
+
+
+# For dispatch
+const DGFD = DG{Basis} where {Basis<:AbstractDerivativeOperator}
+
+
+# 2D containers
+init_mortars(cell_ids, mesh, elements, mortar::Nothing) = nothing
+
+
+# 2D caches
+function create_cache(mesh::TreeMesh{2}, equations,
+                      volume_integral::VolumeIntegralStrongForm, dg, uEltype)
+
+  prototype = Array{SVector{nvariables(equations), uEltype}, ndims(mesh)}(
+    undef, ntuple(_ -> nnodes(dg), ndims(mesh))...)
+  f_threaded = [similar(prototype) for _ in 1:Threads.nthreads()]
+
+  return (; f_threaded,)
+end
+
+create_cache(mesh, equations, mortar::Nothing, uEltype) = NamedTuple()
+nmortars(mortar::Nothing) = 0
+
+
+# 2D RHS
+function calc_volume_integral!(du, u,
+                               mesh::TreeMesh{2},
+                               nonconservative_terms::Val{false}, equations,
+                               volume_integral::VolumeIntegralStrongForm,
+                               dg::DGFD, cache)
+  D = dg.basis # SBP derivative operator
```

Comment:
Is this a kosher use of the basis, i.e., can the SBP derivative op really be considered as "basis"? Or is this just shoehorning the SBP-FD concept into our `DG` type?

Reply:
Hm, still sounds odd to me but I can live with it since I don't have a better solution either.

Reply:
I've wondered about this too - I guess the definition "basis" is somewhat ambiguous. I've usually seen it denote basis functions, but grid functions can be thought of as being constructed from discrete bases too.

Reply:
That's basically what I had in mind. Nevertheless, I'm open to better suggestions if you have any.

Reply:
Not in particular. As an aside, it doesn't look like `LobattoLegendreBasis` really defines basis functions either - it's more like a container for quantities defined on the reference element.

Reply:
Yeah, the Lobatto-Legendre bases we use and the SBP FD operators are very similar from this point of view: They only know that they use nodal representation, have a diagonal mass matrix represented by quadrature weights, and have some associated discrete differentiation operators.

Reply:
We have this as "basis" as well in our Fortran codes. However, I can see, that this is a bit confusing as a name, especially now that we have SBP-FD.
We could replace the name "basis" with something more general: maybe

(i) operator

(ii) discretization

Reply:
`discretization` would IMHO be too general, everything is some sort of discretization or discretization-related, isn't it? I could live with `operator` I guess: it makes it somewhat more general and less intuitive for DG, but my consistency-o-meter says that using a better name for DG does offset a worse name for everything else we want to do with it.

However, I'd leave the change of `basis` to `operator` for another day, create an issue for it, and move on

Reply:
Sounds good to me :+1:

Reply:
Xref #633

Reply:
In David's book this is called `spA` which stands for "spectral approximation". Naming it `operator` sounds weird to me because the struct itself contains several opertors like the D matrix and the diagonal P matrix. I don't have a better suggestion though...